Sources/PackageSets/API.swift

```diff
@@ -0,0 +1,179 @@
+    ///
+    /// A failure is returned if the package is not found.
+    ///
+    /// - Parameters:
+    ///   - reference: The package reference
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke when result becomes available
+    func getPackageMetadata(
+        _ reference: PackageReference,
+        profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<PackageSetsModel.PackageMetadata, Error>) -> Void
+    )
+
+    // MARK: - Target (Module) APIs
+
+    /// List all known targets.
+    ///
+    /// A target name may be found in different packages and/or different versions of a package, and a package
+    /// may belong to multiple package sets. This API's result items will be consolidated by target then package,
+    /// with the package's versions list filtered to only include those that contain the target.
+    ///
+    /// - Parameters:
+    ///   - sets: Optional. If specified, only list targets within these groups.
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke when result becomes available
+    func listTargets(
+        sets: Set<PackageSetsModel.PackageSetIdentifier>?,
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<PackageSetsModel.TargetListResult, Error>) -> Void
+    )
+
+    // MARK: - Search APIs
+
+    /// Finds and returns packages that match the query.
+    ///
+    /// If applicable, for example when we search by package name which might change between versions,
+    /// the versions list in the result will be filtered to only include those matching the query.
+    ///
+    /// - Parameters:
+    ///   - query: The search query
+    ///   - groups: Optional. If specified, only search within these groups.
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke when result becomes available
+    func findPackages(
+        _ query: String,
+        sets: Set<PackageSetsModel.PackageSetIdentifier>?,
+        profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<PackageSetsModel.PackageSearchResult, Error>) -> Void
+    )
+
+    /// Finds targets by name and returns the corresponding packages.
+    ///
+    /// This API's result items will be consolidated by target then package, with the
+    /// package's versions list filtered to only include those that contain the target.
+    ///
+    /// - Parameters:
+    ///   - query: The search query
+    ///   - searchType: Optional. Target names must either match exactly or contain the prefix.
+    ///                 For more flexibility, use the `findPackages` API instead.
+    ///   - groups: Optional. If specified, only search within these groups.
```

Comment:
should say `sets` here, since the parameter has been renamed

Reply:
renamed this to collections and addressed